python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -635,10 +635,14 @@ def _flush(self, objs):
 
         for obj in objs:
             with obj.lock:
-                obj._getWriteAccess()
-                obj_id = getattr(obj, _reg_id_str)
+                # check if the object is dirty, if not do nothing
                 if not obj._dirty:
                     continue
+
+                # flush the object. Need to call _getWriteAccess for consistency reasons
+                # TODO: getWriteAccess should only 'get write access', as that's not needed should it be called here?
+                obj._getWriteAccess()
+                obj_id = getattr(obj, _reg_id_str)
                 self.repository.flush([obj_id])
                 obj._dirty = False
                 self.repository.unlock([obj_id])
```

Comment:
It's only in there because it used to be. I'm happy to remove the call to `unlock` if it's not needed. What is it *supposed* to do?